### PR TITLE
Debug dump

### PR DESCRIPTION
**Pull request by Uitiam**
Debug Dump of All the Entity Coordinates when user is in Debug mode and the press the '.' key

### CODE REVIEW

**Comment by Uitiam** [comment]
Also fixed player being able to drop their default weapon

File: EntityDumpLog.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
+ENTITY POSITION DUMP
```
Comment (JohnAgapeyev) [inline EntityDumpLog.txt]: This file should not be committed to the repo
Add .txt files to the gitignore and remove this file from your branch

File: src/inventory/Inventory.cpp [hunk 2]
```diff
@@ -81,15 +81,17 @@ void Inventory::scrollCurrent(int direction) {
...
-     if (w != nullptr) {
-         if (w->getAmmo() > 0) {
-             GameManager::instance()->createWeaponDrop(x,y, weaponIds[current]);
-
-         } else {
-             //delete weapon From Weapon Manager
-             GameManager::instance()->removeWeapon(weaponIds[current]);
+     if(current){
+         Weapon *w = getCurrent();
+         if (w != nullptr) {
```
Comment (JohnAgapeyev) [inline src/inventory/Inventory.cpp]: `if (w)`

File: src/log/EntityDump.cpp [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+#include <iostream>
+#include "../buildings/DropPoint.h"
+
+
+/*Created by Maitiu Morton 4/3/2017
+ *Loops through each manager and writes there information to
+ *the EntityDumpLog.txt
+ */
+void dumpEntityPositions(const Player* p){
+    #ifndef NDEBUG
```
Comment (JohnAgapeyev) [inline src/log/EntityDump.cpp]: All preprocessor commands should be in the far left column
No indentation

File: src/log/EntityDump.cpp [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    std::ofstream entityDump;
+    std::pair<float, float> coord;
+    GameManager *gm = GameManager::instance();
+
+
+    entityDump.open("EntityDumpLog.txt");
```
Comment (JohnAgapeyev) [inline src/log/EntityDump.cpp]: fstreams can open files in the constructor, please do so

File: src/log/EntityDump.cpp [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+    entityDump << "ENTITY POSITION DUMP\n\n";
+
+    coord  = getCoordinates(dynamic_cast<Entity*>(p->getMarine()));
```
Comment (JohnAgapeyev) [inline src/log/EntityDump.cpp]: No need to separate declaration and initialization

File: src/log/EntityDump.cpp [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    entityDump << "PLAYER MARINE'S POSITION X:" << coord.first << " Y:" << coord.second
+     << std::endl << std::endl;
```
Comment (JohnAgapeyev) [inline src/log/EntityDump.cpp]: Don't use endl as it flushes the stream and I doubt that's what you want judging by how you've used it twice in a row here
It's better to simply write `\n` and flush explicitly

File: src/log/EntityDump.cpp [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+    entityDump << "\nAll Drop POINT POSITIONS:\n";
+    //print Store Positions to File
+    for (const auto& dp : gm->getDropPointManager()) {
+        std::pair<float, float> coord = dp.second.getCoord();
+        entityDump << "Drop Point id:" << dp.first << " Position: " << "X:" << coord.first
+            << " Y:" << coord.second << std::endl;
+    }
+
+    entityDump.close();
```
Comment (JohnAgapeyev) [inline src/log/EntityDump.cpp]: Don't need to close fstreams

File: src/log/EntityDump.cpp [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+/*created by Maitiu Morton 4/3/2017
+ * Gets the Coordinates of an Entity and returns them as a pair
+ */
+std::pair<float, float> getCoordinates(const Entity* e){
+    std::pair<float, float> coord;
+    coord.first = e->getX();
+    coord.second = e->getY();
+    return coord;
```
Comment (JohnAgapeyev) [inline src/log/EntityDump.cpp]: The whole method can become `return {e->getX(), e->getY()};`

**Review by JohnAgapeyev — CHANGES_REQUESTED** [review]

**Review by eyyu — COMMENTED** [review]
can you add a source header please?

**Review by isaacmorneau — APPROVED** [review]

**Review by JohnAgapeyev — APPROVED** [review]

**Pull request merged**